plugin/src/main/java/com/google/errorprone/xplat/checker/LazyInitBan.java

```diff
@@ -0,0 +1,409 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.google.errorprone.xplat.checker;
+
+
+import static com.google.errorprone.BugPattern.SeverityLevel.ERROR;
+
+import com.google.auto.service.AutoService;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.annotations.concurrent.LazyInit;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.bugpatterns.BugChecker.MethodTreeMatcher;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.matchers.Matcher;
+import com.google.errorprone.matchers.Matchers;
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.source.tree.AssignmentTree;
+import com.sun.source.tree.BinaryTree;
+import com.sun.source.tree.BlockTree;
+import com.sun.source.tree.ExpressionStatementTree;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.IdentifierTree;
+import com.sun.source.tree.IfTree;
+import com.sun.source.tree.MethodTree;
+import com.sun.source.tree.ParenthesizedTree;
+import com.sun.source.tree.ReturnTree;
+import com.sun.source.tree.StatementTree;
+import com.sun.source.tree.SynchronizedTree;
+import com.sun.source.tree.Tree.Kind;
+import com.sun.source.tree.VariableTree;
+import com.sun.tools.javac.code.Symbol;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.Modifier;
+import javax.lang.model.element.Name;
+
+@AutoService(BugChecker.class)
+@BugPattern(
+    name = "LazyInitBan",
+    // TODO(lukhnos): Write more detailed explanation and summary, describe why @LazyInit is needed
+    summary = "Lazy Init is an error prone pattern on iOS. Please do it by the book.",
+    explanation =
+        "TBD",
+    severity = ERROR)
+public class LazyInitBan extends BugChecker implements MethodTreeMatcher {
+
+  private static final Matcher<MethodTree> METHOD_MATCHER =
```

Comment:
Perhaps `METHODS_EXCEPT_CONSTURCTOR_MATCHER`, since it's a bit puzzling to remember why a `METHOD_MATCHER` gets immediately used in a `matchMethod()` method below.

Reply:
Done